GZY-441: New tasks not appearing in timer list without manual refresh

vestefano opened the pull request:
Added SocketService to both AutomationTaskSyncHandler and TaskCreateHandler to emit 'tasks:changed' events to employees when tasks are updated. This ensures that all members receive real-time notifications about task changes, enhancing the responsiveness of the task management system.

Card: [New Tasks Not Appearing in Timer List Without Manual Refresh](https://trello.com/c/qME8yAB6/441-new-tasks-not-appearing-in-timer-list-without-manual-refresh)

vestefano committed:
feat: integrate socket service for real-time task updates in automation and task creation handlers

Added SocketService to both AutomationTaskSyncHandler and TaskCreateHandler to emit
'tasks:changed' events to employees when tasks are updated. This ensures that all members
receive real-time notifications about task changes, enhancing the responsiveness of the
task management system.

YanaYestr commented on packages/core/src/lib/tasks/commands/handlers/automation-task.sync.handler.ts:
```diff
@@ -187,6 +189,15 @@ export class AutomationTaskSyncHandler implements ICommandHandler<AutomationTask
 					);
+
+					// Emit tasks:changed event to all members to update their task list in the timer
+					try {
```
These changes are unnecessary. Please remove them, because we currently do not have any integrations with external applications.

YanaYestr commented on packages/core/src/lib/tasks/commands/handlers/automation-task.sync.handler.ts:
```diff
@@ -302,6 +313,16 @@ export class AutomationTaskSyncHandler implements ICommandHandler<AutomationTask
 				}
 			}
 
+			// Emit tasks:changed event to all existing members as well
```
The same - these changes are unnecessary. Please remove them, because we currently do not have any integrations with external applications.

YanaYestr commented on packages/core/src/lib/tasks/commands/handlers/task-create.handler.ts:
```diff
@@ -189,6 +191,15 @@ export class TaskCreateHandler implements ICommandHandler<TaskCreateCommand> {
 					);
+
+					// Emit tasks:changed event to all members to update their task list in the timer
+					try {
```
Please also add a socket when the task is deleted.

vestefano committed:
feat: revert AutomationTaskSyncHandler changes and add task deleted signal dispatching

- Reverted changes in AutomationTaskSyncHandler because cross-platform sync is not used and
was deemed unnecessary for now.
- Added signal dispatching when a task is deleted.
